packages/lib/file-api-driver-dropbox.js

```diff
@@ -171,19 +172,64 @@ class FileApiDriverDropbox {
 		if (typeof content === 'string') content = shim.Buffer.from(content, 'utf8');
 
 		try {
-			await this.api().exec(
-				'POST',
-				'files/upload',
-				content,
-				{
-					'Dropbox-API-Arg': JSON.stringify({
-						path: this.makePath_(path),
-						mode: 'overwrite',
-						mute: true, // Don't send a notification to user since there can be many of these updates
-					}),
-				},
-				options,
-			);
+			if (options && options.source === 'file') {
```

Comment:
sorry, but I don't get you, can you elaborate? I ensured that my approach only opened the file while chunking, hence the change in `uploadBlob`.

Reply:
On line 179 you're reading the full file into memory, even though you only need the first 145 Mb.

Reply:
that's how i would do it, even though it may make the code a bit more complicated

consider if you have a 1GB file, if you read it all into memory you need 1 GB, while reading in chunks requires only 145 Mb

admittedly the example is a bit extreme but on the other hand there were crashes on mobile when the app tried to load attachments of ~100Mb

Reply:
Hey Roman, I added a new commit, when the test passes, take a look... I use a generator to generate chunks. 

the `readBytes` function is there for readability but can be used in place of the call itself with the `await` keyword or the `sync` version.

Reply:
Looks ok to me. Have you tested this on mobile?

Reply:
nope, I'm jammed trying to prepare a proposal... when I finish I'll have the time to test it on mobile and tackle other issues.